### PR TITLE
All pets search endpoint

### LINKED ISSUE #35: GET Pets
Return collection of all pets of a certain type

### PR DESCRIPTION
**Pull request by jakejakearell**
# Modifies
- `app/controllers/api/v1/pets_controller.rb`: Adds index action for search, now only validates pet params when creating a pet, adds a private method for pet type param.
- `app/models/pet.rb`: Add helper class methods to move logic out of controller for search endpoint.
- `config/routes.rb `: Add route for endpoint.
- `spec/models/pets_spec.rb`: Test new class methods.
# Adds
- `spec/requests/api/v1/get_pets_request_spec.rb`: Test data structure of pet search endpoint. Test edge cases and sad paths.

<details>
<summary>Issues</summary>
  closes #35 
<br>

</details>

### CODE REVIEW

File: app/controllers/api/v1/pets_controller.rb [hunk 1]
```diff
@@ -10,12 +10,25 @@ def create
...
+  end
+
   private
 
   def pet_params
     params.permit(:user_id, :name, :age, :pet_type, :breed, :description, :gender, :fixed, :house_trained, :good_with_kids, :good_with_animals, :photo_url_1, :photo_url_2, :photo_url_3)
   end
 
+  def pet_type
+    params.permit(:type)[:type]
```
Comment (arikalea) [inline app/controllers/api/v1/pets_controller.rb:29]: I'm curious why you used :type here instead of pet_type. Is this referring to a different :type?

**Review by arikalea — APPROVED** [review]
Nice work on all of this!
